File: imageviewer/src/main/java/com/king/image/imageviewer/loader/GlideImageLoader.java
package com.king.image.imageviewer.loader;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.king.image.imageviewer.ImageViewer;

import androidx.annotation.Nullable;

/**
 * 当使用{@link GlideImageLoader}时，必须依赖{@link Glide}库。
 * 为了保证{@link ImageViewer}体积最小化，和用户更多可能的选择性，并未将{@link Glide}打包进aar。
 *
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 */
public class GlideImageLoader implements ImageLoader {
    @Override
    public void loadImage(Context context, ImageView imageView,Object data, @Nullable Drawable placeholderDrawable,@Nullable Drawable errorDrawable) {
        Glide.with(context).load(data).placeholder(placeholderDrawable).error(errorDrawable).into(imageView);
    }
}

File: imageviewer/src/main/java/com/king/image/imageviewer/loader/GlideImageLoader.kt
package com.king.image.imageviewer.loader

import android.graphics.drawable.Drawable
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.king.image.imageviewer.ImageDataSource

/**
 * 当使用[GlideImageLoader]时，必须依赖[Glide](https://github.com/bumptech/glide)库
 *
 * @author <a href="mailto:jenly1314@gmail.com">Jenly</a>
 * <p>
 * <a href="https://github.com/jenly1314">Follow me</a>
 */
class GlideImageLoader : ImageLoader {

    override fun loadImage(
        imageView: ImageView,
        model: Any?,
        placeholderDrawable: Drawable?,
        errorDrawable: Drawable?,
    ) {
        if (model is ImageDataSource) {
            loadImage(imageView, model.getDataSource(), placeholderDrawable, errorDrawable)
        } else {
            Glide.with(imageView)
                .load(model)
                .placeholder(placeholderDrawable)
                .error(errorDrawable)
                .into(imageView)
        }
    }
}
